current/advanced/block-import.md

```diff
@@ -0,0 +1,41 @@
+---
+slug: block-import
+lang: en
+title: The Block Import Pipeline
+---
+
+
+## The Import Queue
+
+The import queue is an abstract worker queue present in every Substrate node. It is not part of the runtime. The Import Queue is responsible for processing pieces of incoming information, verifying them, and if they are valid, importing them into the node's state. The most fundamental piece of information that the import queue processes is blocks themselves, but it is also responsible for importing consensus-related messages such as Justifications and, in light clients, finality proofs.
+
+The Import queue collects incoming blocks from the network or from other parts of the local node, and stores them in a pool. The blocks are later checked for validity and discarded if they are not valid. Finally, the block is imported when the import queue is instructed to import it by the consensus engine.
+
+The import queue is codified abstractly in Substrate by means of the [`ImportQueue` trait](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sp_consensus/import_queue/trait.ImportQueue.html). The use of a trait allows each consensus engine to provide its own specialized implementation of the Import Queue which may take advantage of optimization opportunities such as verifying multiple blocks in parallel as they come in across the network.
+
+## The Basic Queue
+Substrate also provides a default implementation known as the [`BasicQueue`](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sp_consensus/import_queue/struct.BasicQueue.html). The `BasicQueue` does not do any kind of optimization, rather it performs the verification and import steps sequentially. It does, however, abstract the notion of verification through the use of the `Verifier`(https://substrate.dev/rustdocs/v2.0.0-alpha.8/sp_consensus/import_queue/trait.Verifier.html) trait.
+
+Any consensus engine that relies on the `BasicQueue` must implement the `Verifier` trait. In the simplest cases, such as Sunstrate's instant seal consensus, the `Verifier` is simple declares all blocks as valid. In slot-based engines check that the blocks timestamp actually falls in the correct time window for the slot it is claiming.
```

Comment:
```suggestion
Any consensus engine that relies on the `BasicQueue` must implement the `Verifier` trait. In the simplest cases, such as Substrate's instant seal consensus, the `Verifier` is simple declares all blocks as valid. In slot-based engines check that the blocks timestamp actually falls in the correct time window for the slot it is claiming.
```

It might be easier to just say that we verify inherent data in the verifier, or that we verify that the block has been signed by the expected authority.